R/knit_print.R

```diff
@@ -7,13 +7,37 @@
 #'
 #' @keywords internal
 #' @noRd
-knit_print.fontawesome <- function(x, ...) {
+knit_print.fontawesome <- function(x, ..., options, inline = FALSE) {
 
   # nocov start
 
   if (requireNamespace("knitr", quietly = TRUE)) {
-    knitr::knit_print(as.character(x), ...)
+    if (knitr::is_html_output()) {
+      NextMethod()
+    } else if (knitr::pandoc_to(c("latex", "beamer", "docx"))) {
+      if (!requireNamespace("rsvg", quietly = TRUE)) {
+        stop("Using fontawesome with non HTML output requires the rsvg package:\n",
+             " * It can be installed with `install.packages(\"rsvg\")`.",
+             call. = FALSE)
+      }
+      formats <- switch(knitr::pandoc_to(),
+                        beamer = ,
+                        latex = list(ext = ".pdf", renderer = rsvg::rsvg_pdf),
+                        docx = list(ext = ".png", renderer = rsvg::rsvg_png)
```

Comment:
Just curious: what are the pixel dimensions generated by `rsvg_png()`? I have two concerns in mind:

* We don't want the files to be too large
* If there's a large icon, will it look pixellated or blurry?

I should also add that these aren't hugely important, but it would be good to make sure that there's reasonable behavior.

Reply:
I think it is the same for PDF file produced by `rsvg_pdf()` - I did not set any size in the function.

For PNG, png file is 3.5kb in my test. 
For PDF, pdf file is 1kb in my test.

@rich-iannone I would like help on that as you may be more used to this than me. What is your advice for size ? Are the default of **rsvg** functions ok ?

Reply:
I think we can use smaller pixel dimensions.

When I use the `r-project` icon, the file size is 15kB, which is pretty large:

```R
library(fontawesome)
library(rsvg)

name <- "r-project"
rsvg_png(charToRaw(as.character(fa(name))), "icon.png")
file.size("icon.png")
#> [1] 15454
dim(png::readPNG("icon.png"))
#> [1] 512 581   4
```

I believe that the pixel dimensions are taken from the `viewBox` value from the SVG:

```R
fontawesome:::fa_tbl$svg[fontawesome:::fa_tbl$name == name]
#> [1] "<svg viewBox=\"0 0 581 512\"> ...
```

If we make the icon 1/2 the width and height, results in a 256x290 image. For printing, my understanding is that 300ppi is the standard for high-quality prints. That means that the icon could be almost 1 inch in size and still meet that standard.

```R
# Generate icon at half size
width <- fontawesome:::fa_tbl$width[fontawesome:::fa_tbl$name == name]
rsvg_png(charToRaw(as.character(fa(name))), "icon.png", width = width/2)
file.size("icon.png")
#> [1] 6831
dim(png::readPNG("icon.png"))
#> [1] 256 290   4
```

If users really need a giant icon, I suppose we could add a parameter that allowed them to control the output PNG size, but I don't think that's necessary at this point.